dotCMS/src/main/java/com/dotmarketing/common/reindex/ReindexQueueAPIImpl.java

```diff
@@ -88,6 +88,29 @@ public boolean areRecordsLeftToIndex() throws DotDataException {
     public long recordsInQueue() throws DotDataException {
         return recordsInQueue(DbConnectionFactory.getConnection());
     }
+
+    @Override
+    public boolean waitForEmptyQueue(int maxWaitSeconds) throws DotDataException {
+        // Uses Busy Waiting with polling of DB every second to check if there are records in the queue
+        // This should be replaced with a more efficient solution that uses a notification mechanism
+        // but that would require a more complex implementation and this is required for some tests currently
+        try {
+            long recordsInQueue = recordsInQueue();
```

Comment:
this is ok, but wondering if we cannot eventually change this kind of infinite stuff to Stream or Iterable